corehq/apps/sms/tasks.py

```diff
@@ -20,29 +23,50 @@
 from threading import Thread
 
 
+def remove_from_queue(queued_sms):
+    with transaction.atomic():
+        sms = SMS()
+        for field in sms._meta.fields:
+            if field.name != 'id':
+                setattr(sms, field.name, getattr(queued_sms, field.name))
+        queued_sms.delete(sync_to_couch=False)  # Remove sync_to_couch when SMSLog is removed
+        sms.save()
+
+    if sms.direction == OUTGOING and sms.processed and not sms.error:
+        create_billable_for_sms(sms)
+    elif sms.direction == INCOMING and sms.domain and domain_has_privilege(sms.domain, privileges.INBOUND_SMS):
+        create_billable_for_sms(sms)
+
```

Comment:
@nickpell fyi, i updated where create_billable_for_sms gets called and the conditions under which it gets called. let me know if you're good with this (can just search for create_billable_for_sms in the diff)